pkg/events/redis.go

```diff
@@ -242,6 +243,12 @@ func (r *RedisConsumer) consumer(ctx context.Context, stream string, ch reflect.
 			messages, err := r.pop(con, group, stream)
 			if err != nil {
 				logger.Error().Err(err).Msg("failed to get events from")
+				select {
+				case <-ctx.Done():
+					return
+				case <-time.After(2 * time.Second):
```

Comment:
What is the significance of 2 seconds in here? why not `default: continue` ?

Reply:
it's just a prevention to avoid getting stuck in a busy loop. I was thinking of differentiate also between normal error and connection errors and in the later case the entire loop should exit because that's a fatal error.